iOS13NativeSwiftTemplate/iOS13NativeSwiftTemplate/AppDelegate.swift

```diff
@@ -57,7 +57,20 @@ class AppDelegate : UIResponder, UIApplicationDelegate {
         return true
     }
 
-    func registerForRemotePushNotifications() {        PushNotificationManager.sharedInstance().registerForRemoteNotifications();
+    func registerForRemotePushNotifications() {
+        UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound, .badge]) { (granted, error) in
+            if granted {
+                DispatchQueue.main.async {
+                    PushNotificationManager.sharedInstance().registerForRemoteNotifications()
+                }
+            } else {
+                SalesforceLogger.d(AppDelegate.self, message: "Push notification authorization denied")
+            }
+
+            if let error = error {
```

Comment:
From the description of `requestAuthorization` I couldn’t tell if it was possible to be granted and also error (maybe granted for one option but errors on another?) so I didn’t stop registration if there was an error